referenceframe/ptgFrame.go

```diff
@@ -0,0 +1,166 @@
+package referenceframe
+
+import (
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/spatialmath"
+)
+
+const (
+	defaultSimDistMM      = 1100.
+	defaultAlphaCnt  uint = 121
+)
+
+const (
+	ptgIndex int = iota
+	trajectoryIndexWithinPTG
+	distanceAlongTrajectoryIndex
+)
+
+type ptgFactory func(float64, float64, float64) tpspace.PrecomputePTG
+
+var defaultPTGs = []ptgFactory{
+	tpspace.NewCirclePTG,
+	tpspace.NewCCPTG,
+	tpspace.NewCCSPTG,
+	tpspace.NewCSPTG,
+	tpspace.NewAlphaPTG,
+}
+
+type ptgGridSimFrame struct {
+	name       string
+	limits     []Limit
+	geometries []spatialmath.Geometry
+	ptgs       []tpspace.PTG
+}
+
+// NewPTGFrameFromTurningRadius will create a new Frame which is also a tpspace.PTGProvider. It will precompute the default set of
+// trajectories out to a given distance, or a default distance if the given distance is <= 0.
+func NewPTGFrameFromTurningRadius(name string, velocityMMps, turnRadMeters, simDist float64, geoms []spatialmath.Geometry) (Frame, error) {
+	if velocityMMps <= 0 {
+		return nil, fmt.Errorf("cannot create ptg frame, movement velocity %f must be >0", velocityMMps)
+	}
+	if turnRadMeters <= 0 {
+		return nil, fmt.Errorf("cannot create ptg frame, turning radius %f must be >0", turnRadMeters)
+	}
+
+	if simDist <= 0 {
+		simDist = defaultSimDistMM
+	}
+
+	// Get max angular velocity in radians per second
+	maxRadps := velocityMMps / (1000. * turnRadMeters)
+	pf := &ptgGridSimFrame{name: name}
+	err := pf.initPTGs(velocityMMps, maxRadps, simDist)
+	if err != nil {
+		return nil, err
+	}
+
+	pf.geometries = geoms
+
+	pf.limits = []Limit{
+		{Min: 0, Max: float64(len(pf.ptgs) - 1)},
+		{Min: 0, Max: float64(defaultAlphaCnt)},
+		{Min: 0},
```

Comment:
This is the only time we've made a limit with no Max, can you manually write it in here?  To me the implication is that there is no max but Go will just fill it with a zero.  This seems really odd to me though and I'm wondering what this is supposed to represent